src/EditorFeatures/VisualBasicTest/TypeInferrer/TypeInferrerTests.vb

```diff
@@ -289,8 +289,14 @@ End Class</text>.Value
 
         <Fact, Trait(Traits.Feature, Traits.Features.TypeInferenceService)>
         <WorkItem(542837)>
-        Public Sub TestArrayRankSpecifier()
-            TestInMethod("Dim q As String() = New String([|Foo()|])", "System.Object")
+        Public Sub TestArrayRankSpecifier1()
+            TestInMethod("Dim q As String() = New String([|Foo()|])", "System.Char()")
+        End Sub
+
+        <Fact, Trait(Traits.Feature, Traits.Features.TypeInferenceService)>
+        <WorkItem(542837)>
+        Public Sub TestArrayRankSpecifier2()
+            TestInMethod("Dim q As String() = New String([|Foo()|]) { }", "System.Int32")
```

Comment:
Here we actually test creating an array.  And we properly get int32 inferred here.